rust/arrow-flight/src/utils.rs

```diff
@@ -129,21 +129,29 @@ impl TryFrom<&SchemaResult> for Schema {
 pub fn flight_data_to_arrow_batch(
     data: &FlightData,
     schema: SchemaRef,
-) -> Result<Option<RecordBatch>> {
+) -> Option<Result<RecordBatch>> {
```

Comment:
Is the reason for this change because a Rust iterator has to return an `Option`  (so it can signal none)?

Reply:
Yes, that is the reason. we could [`transpose`](https://doc.rust-lang.org/std/result/enum.Result.html#method.transpose) these operations elsewhere, but the gist is that the iterator `next` needs the `Option` to come first (which is not very user-friendly as it limits the usage of `?`)